src/Cli/dotnet/commands/dotnet-sln/add/Program.cs

```diff
@@ -2,143 +2,183 @@
+        private void AddProjectWithDefaultGuid(SolutionModel solution, string relativePath, SolutionFolderModel solutionFolder)
         {
-            return !path.StartsWith("..");
-        }
+            // Open project instance to see if it is a valid project
+            ProjectRootElement projectRootElement = ProjectRootElement.Open(relativePath);
+            SolutionProjectModel project;
+            try
+            {
+                project = solution.AddProject(relativePath, null, solutionFolder);
+            }
+            catch (SolutionArgumentException ex) when (ex.ParamName == "projectTypeName")
+            {
+                // If guid is not identified by vs-solutionpersistence, check in project element itself
+                var guid = projectRootElement.GetProjectTypeGuid();
+                if (string.IsNullOrEmpty(guid))
+                {
+                    Reporter.Error.WriteLine(CommonLocalizableStrings.UnsupportedProjectType, Path.GetFullPath(relativePath));
+                    return;
+                }
+                project = solution.AddProject(relativePath, guid, solutionFolder);
+            }
+            // Generate intermediate solution folders
+            if (solutionFolder is null && !_inRoot)
+            {
+                var relativePathDirectory = Path.GetDirectoryName(relativePath);
+                if (!string.IsNullOrEmpty(relativePathDirectory))
+                {
+                    SolutionFolderModel relativeSolutionFolder = solution.AddFolder(GetSolutionFolderPathWithForwardSlashes(relativePathDirectory));
+                    project.MoveToFolder(relativeSolutionFolder);
+                    // Avoid duplicate folder/project names
+                    if (project.Parent is not null && project.Parent.ActualDisplayName == project.ActualDisplayName)
+                    {
+                        solution.RemoveFolder(project.Parent);
+                    }
+                }                
+            }
+            // Add settings based on existing project instance
+            ProjectInstance projectInstance = new ProjectInstance(projectRootElement);
+            string projectInstanceId = projectInstance.GetProjectId();
+            if (!string.IsNullOrEmpty(projectInstanceId))
+            {
+                project.Id = new Guid(projectInstanceId);
+            }
+            var projectInstanceBuildTypes = projectInstance.GetConfigurations();
+            var projectInstancePlatforms = projectInstance.GetPlatforms();
 
-        private static string TrimProject(string path)
-        {
-            return Path.GetDirectoryName(path);
-        }
+            foreach (var solutionPlatform in solution.Platforms)
+            {
+                var projectPlatform = projectInstancePlatforms.FirstOrDefault(
+                    x => x.Replace(" ", string.Empty) ==  solutionPlatform.Replace(" ", string.Empty), projectInstancePlatforms.FirstOrDefault("Any CPU"));
```

Comment:
The current behavior expects to match platform regardless of whitespaces, so `AnyCPU == AnyCPU`. For the majority of cases this shouldn't have additional consequences.